src/main/java/com/example/weather/client/exceptions/MyResourceNotFoundException.java

```diff
@@ -0,0 +1,12 @@
+package com.example.weather.client.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.NOT_FOUND)
+public class MyResourceNotFoundException extends RuntimeException {
```

Comment:
Im not sure the name of the class of HTTP status is suitable... 
We're throwing this exception if page number is invalid.
Besides this, that exception should also be thrown if the `size` parameter is too big.
If the exception is being thrown for invalid parameter values better would be invalid request HTTP status.